tests/cli.rs

```diff
@@ -19,6 +22,36 @@ fn create_output_dir() {
     .success();
 
     File::open("tests/create_me/flipper.json").expect("should exist");
```

Comment:
Can we please use something like the `tempdir` from [tempfile](https://crates.io/crates/tempfile) instead? If the compiler crashes during the test, e.g. because of tinkering in codegen or emit. Then these files just sit there and show up in git status and are easily committed by accident (`git add tests` :eyes:). Where as with using `tempfile` worst case the files just end up hang around in some location I don't see and don't care about.

As a bonus, with `tempfile` we don't have to deal with deletion of these files manually.

Reply:
We are testing directory creation as well as file creation. The output is ignored by `.gitignore` so nothing should go into `git status` or `git add tests/`.

Reply:
Ok this is siightly defused, the reason why I saw this before is that the `.json` files were not in `.gitignore`.

However I still think it would the right approach to use a `tempdir` as the directory creation can be tested inside there as well. Not a fan of abusing our `tests` dir as a `/tmp`

Reply:
I see your point, I don't mind adding tempdir for this.

My personal issue with tempdir is that it not really a panacea, if your program exits with e.g. `std::process::exit(1)` you still have a temp directory with garbage, but now its in /tmp. There are some tricks you can do with creating a single files without a name but that does not help for our case, because we need to test a directory.

Reply:
ok, the test now uses the `tempfile` crate (the tempdir crate docs says to use tempfile instead). It's actually pretty nice and now the test is shorter in code, because the cleanup is much simpler, and now in nearly all cases the temp directory is gone. Definitely an improvement, thanks.